### PR TITLE
Use NamedWatcher in Agent's k8s provider

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
Replacing `NewWatcher` function and use `NewNamedWatcher` in its place.

## Why is it important?
Code's maintainability.

## Related issues

- Closes https://github.com/elastic/beats/issues/28780

### CODE REVIEW

File: libbeat/common/kubernetes/watcher.go [hunk 1]
```diff
@@ -93,12 +93,6 @@ type watcher struct {
 	logger   *logp.Logger
 }
 
-// NewWatcher initializes the watcher client to provide a events handler for
-// resource from the cluster (filtered to the given node)
-func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOptions, indexers cache.Indexers) (Watcher, error) {
-	return NewNamedWatcher("", client, resource, opts, indexers)
-}
-
 // NewNamedWatcher does the same as NewWatcher, but also allows to name the k8s
```
Comment (MichaelKatsoulis) [inline libbeat/common/kubernetes/watcher.go:96]: Nit: After deleting `NewWatcher` this sentence does not make sense any more

**Review by MichaelKatsoulis — APPROVED** [review]
LGTM! Just one small update in the `NewNamedWatcher` description.

**Review by tetianakravchenko — APPROVED** [review]
LGTM

**Pull request merged**